### PR TITLE
[C#]Remove null checks for C# value types

### PR DESCRIPTION
**Pull request by etherealjoy**
### PR checklist

- [X] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [X] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`. If contributing template-only or documentation-only changes which will change sample output, be sure to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) first.
- [X] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.1.x`, `5.0.x`. Default: `master`.
- [X] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

Remove null checks on C# value types which throws a lot of warnings.

Null checks on enums which is a value type are already avoided. Combining simple types and enum as value types and avoiding null checks.
See here.
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/value-types

@wing328 @mandrean @jimschubert

### CODE REVIEW

**Comment by etherealjoy** [comment]
I will regenerate code if it looks OK

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractCSharpCodegen.java [hunk 1]
```diff
@@ -454,6 +461,19 @@ private void postProcessEnumRefs(final Map<String, Object> models) {
                         var.isPrimitiveType = true;
                     }
                 }
+                for (CodegenProperty var : model.vars) {
```
Comment (etherealjoy) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractCSharpCodegen.java]: Question:
why is this line (some lines above) not covering all `enums`
```
for (CodegenProperty var : model.allVars) 
```

Comment (wing328) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractCSharpCodegen.java]: It should. That could be a bug with allVars.

I'll look into it in the coming week.

**Comment by etherealjoy** [comment]
I think I need to add extra condition if it is a `valueType` but not `nullable`, however I see parsing of the nullable field is not done.

**Comment by wing328** [comment]
You mean something like `{{#isNullable}} ... {{/isNulalble}}`?

**Comment by etherealjoy** [comment]
Exactly. So if the value is nullable it will not be e.g. `int` but `int?`. But in this case if the `dataType` is used for postprocess it is not anymore a `valueType`
I will try out some specs with `nullable` fields.

**Comment by wing328** [comment]
I think nullable is supported in the C# generators already: https://github.com/OpenAPITools/openapi-generator/pulls?utf8=%E2%9C%93&q=is%3Apr+is%3Amerged+nullable+csharp

 but you will need to use `x-nullable` in OAS v2 instead which does not support nullable natively.

**Comment by etherealjoy** [comment]
Thank you. I see that the simple type is already changed to nullable type. Then this will be covered.
I will make the test with this.

**Comment by etherealjoy** [comment]
Ok, I verified it. the nullable types are having null checks. only value Types are skipped.

**Comment by wing328** [comment]
Can you please update the C# netcore sample as well?
```
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   samples/client/petstore/csharp-netcore/OpenAPIClient/docs/EnumTest.md
	modified:   samples/client/petstore/csharp-netcore/OpenAPIClient/src/Org.OpenAPITools/Model/ApiResponse.cs
	modified:   samples/client/petstore/csharp-netcore/OpenAPIClient/src/Org.OpenAPITools/Model/Cat.cs
	modified:   samples/client/petstore/csharp-netcore/OpenAPIClient/src/Org.OpenAPITools/Model/CatAllOf.cs
	modified:   samples/client/petstore/csharp-netcore/OpenAPIClient/src/Org.OpenAPITools/Model/Category.cs
	modified:   samples/client/petstore/csharp-netcore/OpenAPIClient/src/Org.OpenAPITools/Model/EnumTest.cs
	modified:   samples/client/petstore/csharp-netcore/OpenAPIClient/src/Org.OpenAPITools/Model/FormatTest.cs
```

**Pull request merged**